src/MobileDevice/widget/MobileDevice.ts

```diff
@@ -0,0 +1,164 @@
+        }
+    }
+
+    private setUpEvents() {
+        if (window.device) {
+            this.deviceReadyEvent = dojoOn(document, "deviceReady", dojoLang.hitch(this, this.onDeviceReady));
+        }
+    }
+
+    private removeEvents() {
+        if (window.device && this.deviceReadyEvent) {
+            this.deviceReadyEvent.remove();
+            this.deviceReadyEvent = null;
+        }
+    }
+
+    private setUpWidgetDom() {
+        domConstruct.create("div", {
+            class: "mobile-device-widget",
+            id: "mobile-device"
+        }, this.domNode, "first");
+    }
+
+    private onDeviceReady() {
+        this.getAppInformation();
+    }
+
+    private validateProps(): string {
+        let errorMessage = "";
+        if (this.onDeviceReadyAction === "callMicroflow" && !this.microflow) {
+            errorMessage = "on click microflow is required in the 'Events' tab, 'Microflow' property";
+        } else if (this.onDeviceReadyAction === "showPage" && !this.page) {
+            errorMessage = "on click page is required in the 'Events' tab, 'Page' property";
+        }
+
+        return errorMessage && `Error in mobile device widget configuration: ${errorMessage}`;
+    }
+
+    private setDeviceInformation() {
+        if (this.deviceTypeAttribute) {
+            this.mxObject.set(this.deviceTypeAttribute, window.device ? window.device.platform : "Web");
+        }
+        if (this.deviceIdAttribute) {
+            this.mxObject.set(this.deviceIdAttribute, window.device ? window.device.uuid : "");
+        }
+    }
+
+    private getWebDeviceInformation() {
+        if (!window.device) {
+            if (this.onNavigateBack) {
+                mx.ui.back();
+            }
+            this.executeMicroFlow(this.mxObject);
+        }
+    }
+
+    private getAppInformation() {
+        if (cordova.hasOwnProperty("getAppVersion")) {
+            cordova.getAppVersion.getAppName(appName => {
+                if (this.appVersionNameAttribute) {
+                    this.mxObject.set(this.appVersionNameAttribute, appName);
+                }
+                cordova.getAppVersion.getPackageName(packageName => {
+                    if (this.appVersionIdAttribute) {
+                        this.mxObject.set(this.appVersionIdAttribute, packageName);
+                    }
+                    cordova.getAppVersion.getVersionNumber(versionNumber => {
+                        if (this.appVersionVersionAttribute) {
+                            this.mxObject.set(this.appVersionVersionAttribute, versionNumber);
+                        }
+                        if (this.onNavigateBack) {
+                            mx.ui.back();
+                        }
+                        this.executeMicroFlow(this.mxObject);
+                    });
+                });
+            });
+        } else {
+            window.mx.ui.error("Mobile device widget: add cordova-plugin-app-version to your project config");
```

Comment:
Only show error when you expect the plugin the be needed.
`if (this.appVersionNameAttribute || this.appVersionIdAttribute || this.appVersionVersionAttribute) {`